### DISCUSSION TITLE
Use of malloc

### DISCUSSION THREAD

**Discussion by tba10596** [funbiscuit/embedded-cli#36] [General]
in embedded_cli setup, malloc is used because it "guarantees alignment". Is there another way to do this so that heap can be set to zero?

**Comment by funbiscuit** [comment]
I think I don't understand question. If you want to set your buffer to zero, you can use memset and pass 0. But that's not needed, since it's already done:
https://github.com/funbiscuit/embedded-cli/blob/master/lib/src/embedded_cli.c#L419

If you can't use malloc, you can just declare static buffer of memory and pass it to cli.
